SearchWord.java

```diff
@@ -0,0 +1,59 @@
+import java.util.LinkedList;
+
+public class SearchWord {
+
+	public int[][] visited;
+
+	public LinkedList<String> wordSearch(int rows, int columns, char[][] characters, Dictionary dict) {
+
+		LinkedList<String> validWords = new LinkedList<String>();
+		visited = new int[rows][columns];
+
+		for (int i = 0; i < rows; i++) {
+			for (int j = 0; j < columns; j++) {
+				visited[i][j] = 0;
+			}
+		}
+
+		for (int i = 0; i < rows; i++) {
+			for (int j = 0; j < columns; j++) {
+				applyBFS(rows, columns, i, j, characters, dict, validWords, "");
+			}
+		}
+
+		return validWords;
+	}
+
+	public void applyBFS(int rows, int columns, int posX, int posY, char[][] characters, Dictionary dict,
+			LinkedList<String> validWords, String word) {

```

Comment:
I would say contains(word) has O(n) complexity and that of add(word) O(1). A HashSet is better, as both contains and add have O(1) complexity.